atlasdb-tests-shared/src/main/java/com/palantir/atlasdb/transaction/impl/AbstractSerializableTransactionTest.java

```diff
@@ -1083,8 +1084,12 @@ public void testMarkTableInvolvedDoesNoCheckLocksForExpiryOnCommitWhenNotRequire
         // Test table is not thorough, therefore no immutable timestamp lock checking
         testMarkTableInvolvedLockChecksForExpiryOnCommit(TEST_TABLE);
 
-        // Test table is thorough, but still no immutable timestamp lock checking
-        testMarkTableInvolvedLockChecksForExpiryOnCommit(TEST_TABLE_THOROUGH);
+        if (keyValueService.sweepsEntriesInStrictlyNonDecreasingFashion()) {
+            testMarkTableInvolvedLockChecksForExpiryOnCommit(TEST_TABLE_THOROUGH);
```

Comment:
This is probably a bit of an artifact of older atlas code, but let's use the `assertThatCode(() -> ...).doesNotThrowAnyException();` here to contrast the other path.